basicLibrary/src/main/java/basicLibrary/Library.java

```diff
@@ -71,5 +62,50 @@ public static double getAverage(int[] arr) {
         return bestSoFar;
     }
 
+    public static String analyzeWeatherData(int[][] weeklyMonthTemperatures) {
+        int lowestTemp = weeklyMonthTemperatures[0][0];
+        int highestTemp = weeklyMonthTemperatures[0][0];
+
+        // HashSet
+        HashSet<Integer> temps = new HashSet<>();
+
+        // Iterate through a data set and fill HashSet
+        for (int[] weeklyMonthTemperature : weeklyMonthTemperatures) {
+            for (int i : weeklyMonthTemperature) {
```

Comment:
I worry that `i` is a confusing variable name. `i` is conventionally used for index. In this case `i` is an actual piece of weather data, not the index of that data